driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1162,6 +1162,8 @@ private Manager(String clusterName, List<InetSocketAddress> contactPoints, Confi
 
             this.metrics = configuration.getMetricsOptions() == null ? null : new Metrics(this);
             this.listeners = new CopyOnWriteArraySet<Host.StateListener>(listeners);
+
+            this.scheduledTasksExecutor.scheduleWithFixedDelay(new CleanupIdleConnectionsTask(), 10, 10, TimeUnit.SECONDS);
```

Comment:
On the other hand, we also want to avoid creating too many executors (this was something we debated when adding the connection reaper). scheduledTasksExecutor is used to introduce a delay in ADD/UP notifications, which should not happen too frequently. The worst that can happen in these is a connection timeout, which is about 10 seconds by default. If a couple of notifications are stuck in the queue, that would delay the cleanup tasks by ~20 seconds, and the timeout has a granularity of 10 seconds. Overall I think this is acceptable, what do you think?